src/components/post.js

```diff
@@ -1,69 +1,123 @@
+    this.post = this.props.post;
+    this.user = this.props.user;
+  }
 
-export function Post(props) {
-  return (
-    <View>
-      <View
-        style={{
-          flexDirection: 'row',
-          borderBottomColor: '#6a717d',
-          borderBottomWidth: 1,
-        }}>
-        <View style={{alignItems: 'center', flex: 1, padding: 5}}>
-          <Avatar
-            size="medium"
-            rounded
-            source={{uri: props.avatar}}
-            onPress={props.avatarOnPress}
-          />
-          <Text style={{marginTop: 10}}>{props.username}</Text>
-        </View>
+  isLiked = () => {
+    const {usersLike} = this.props.post;
+    const likeIndex = usersLike.findIndex(item => {
+      return item == this.props.user.id;
+    });
+    if (likeIndex != -1) return true;
+      else return false;
+  };
 
-        <View
-          style={{
-            flex: 4,
-            margin: 5,
-            padding: 5,
-            flexDirection: 'column',
-            justifyContent: 'space-between',
-          }}>
-          <Text>{props.postText}</Text>
-          <Text>
-            `
-            {props.postImage && (
-              <Image
-                style={{width: 200, height: 200}}
-                source={{uri: props.postImage}}
-              />
-            )}
-          </Text>
-          <View style={{flexDirection: 'row'}}>
-            <Text style={{flex: 4, textAlign: 'center', alignSelf: 'flex-end'}}>
-              {props.postTime}
-            </Text>
-            <CheckBox
-              containerStyle={{
-                flex: 1,
-                padding: 0,
-                margin: 0,
-                alignSelf: 'flex-end',
+  navigate = (name, params) => () => {
+    NavigationService.navigate(name, params);
+  };
+
+  like = () => {
+    this.setState({liked: !this.state.liked});
+    this.props.likePost(this.user.id, this.post.postId);
+  };
+  componentDidMount(){
+    this.setState({liked : this.isLiked()})
+  }
+
+  render() {
+    return (
+      <View>
+        <View style={styles.postContainer}>
+          <View style={styles.postAvatar}>
+            <Avatar
+              size="medium"
+              rounded
+              source={{
+                uri: lodash.get(this.post, 'user.avatarUri', image.AVATAR),
               }}
-              uncheckedIcon={
-                <Icon name="favorite-border" reverseColor="red" color="red" />
-              }
-              checkedIcon={
-                <Icon name="favorite" reverseColor="red" color="red" />
-              }
-              checked={true}
+              onPress={this.navigate('Profile', {user: this.post.user})}
             />
+            <Text style={styles.postUsername}>
+              {lodash.get(this.post, 'user.username', values.DEFAULT_USERNAME)}
+            </Text>
+          </View>
+
+          <View style={styles.postContentContainer}>
+            <Text>
+              {this.post.postText && <Text>{this.post.postText}</Text>}
```

Comment:
<Text>{this.post.postText}</Text> use lodash for get **postText**. Text tag will not be displayed if it receives an indefinite